### ISSUE TITLE
Suggestion: Allow setting of task-id and/or alias via the web interface

### ISSUE THREAD

**Issue by pjf** [HabitRPG/habitica#683]
When creating new habits via the API, we can set the task-id to any string we like. This is awesome, because with a task id like `junkfood` or `exercise` it's easy to refer to habits later on. A command-line client can do something like `hrpg +exercise` and it's pretty obvious what's going on. Short names are essential if we want to be able to update habits via SMS gateway, or anywhere else that space is limited.

However the web interface only assigns a UUID when creating a task, which makes referring to those tasks later more challenging. If an end client wants to map between human-readable short-labels and UUIDs, then those mappings end up being unique to that application, and need to be set up for each end-point.

There are a couple of ways I can see of getting around this. I'm assuming for the moment that task-ids themselves shouldn't change:
1. Allow users to assign an optional task-id when creating a task via the web interface. However this doesn't help with existing tasks, or if a user later decides they want to set the task-id.
2. Provide the facility for users to set a short task nickname, or alias. Change the API so that if we fail to access a task by task-id, we try via alias instead. This allows for aliases to be shared between applications, and also allows them to be set after task creation.

Obviously I think that the second option is way nicer. :)
##

**Comment by rdhammond** [comment]
I think the second version is the way to go, too. As much as I hate creating two entry points, it prevents breaking backwards compatibility.

Mongo should be able to do the heavy lifting across all the players' tasks, so this seems like a good starting point for my first bugfix :)

**Comment by Alys** [comment]
I like the idea of an alias (I'd use it myself VERY frequently) but @rdhammond I think we should hear opinions from staff and other developers before you work on this, otherwise you might find that your work is never used. Our top developers will receive this message (and yours) in an email.

My initial gut reaction is that the extra complication could make our code less reliable - I might well be wrong, but this would be best discussed before it's implemented. And about being able to do heavy lifting... the site already does some pretty heavy lifting very often, and we get glitches when it's too heavy. I'd be reluctant to add to that. Also note that this is a feature request, not a bug fix, so while it would be nice to have we might end up deciding that it could cause more problems than it solves.

Having said all that, I hope we decide that it won't create problems, because it would be nice! :)

As a less convenient partial alternative, the [Data Display Tool](https://oldgods.net/habitrpg/habitrpg_user_data_display.html) will show you the ID for each of your tasks (Task Overview section, click on "toggle developer data").

**Comment by rdhammond** [comment]
Oh okay. Mostly I'm just raring to go and grabbing the oldest things off the minor changes I can find. :) I'll filter out feature requests for the time being.

**Comment by deilann** [comment]
Thank you for thinking to do that! I've been trying to do more labeling, so
folks can do exactly that. It just gets so repetitious!

On Thu, Mar 12, 2015 at 8:15 PM, R. Daniel Hammond <notifications@github.com

**Comment by Alys** [comment]
@rdhammond I like the way you are thinking! :) Although to be honest, the newer issues are probably more relevant to us now. You are most welcome to work from either end of the queue, but don't feel that the older ones need to be completed first!

Also, there's no need to filter out feature requests in general. Several of the recent feature requests in github are things we very much want. It's just that this one particular feature request causes me concern. So as a general rule: don't bother trying to distinguish between feature requests and bug fixes. :)

**Comment by Alys** [comment]
@rdhammond Also, there's no need to filter out feature requests in general. Several of the recent feature requests in github are things we very much want. It's just that this one particular feature request causes me concern. So as a general rule: don't bother trying to distinguish between feature requests and bug fixes. :)

**Comment by crookedneighbor** [comment]
Do to some breaking changes coming up with V3, we can't consider this feature request at this time.
